src/main/java/org/opensearch/performanceanalyzer/rca/store/rca/admissioncontrol/AdmissionControlRca.java

```diff
@@ -152,50 +136,18 @@ public ResourceFlowUnit operate() {
                 new HotNodeSummary(
                         instanceDetails.getInstanceId(), instanceDetails.getInstanceIp());
 
-        HeapMetrics heapMetrics = getHeapMetric();
-        if (heapMetrics.usedHeap == 0 || heapMetrics.maxHeap == 0) {
-            ResourceContext context = new ResourceContext(HEALTHY);
-            return new ResourceFlowUnit<>(currentTimeMillis, context, nodeSummary);
-        }
-
-        double currentHeapPercent = (heapMetrics.usedHeap / heapMetrics.maxHeap) * 100;
-
-        // If we observe heap percent range change then we tune request-size controller threshold
-        // by marking resource as unhealthy and setting desired value as configured
-        if (requestSizeHeapRange.hasRangeChanged(previousHeapPercent, currentHeapPercent)) {
-            double desiredThreshold = getHeapBasedThreshold(currentHeapPercent);
-            if (desiredThreshold == 0) {
-                // AdmissionControl rejects all requests if threshold is set to 0, thus ignoring
-                ResourceContext context = new ResourceContext(HEALTHY);
-                return new ResourceFlowUnit<>(currentTimeMillis, context, nodeSummary);
-            }
-            LOG.debug(
-                    "[AdmissionControl] Observed range change. previousHeapPercent={} currentHeapPercent={} desiredThreshold={}",
-                    previousHeapPercent,
-                    currentHeapPercent,
-                    desiredThreshold);
-
-            previousHeapPercent = currentHeapPercent;
-
-            HotResourceSummary resourceSummary =
-                    new HotResourceSummary(HEAP_MAX_SIZE, desiredThreshold, currentHeapPercent, 0);
-            nodeSummary.appendNestedSummary(resourceSummary);
-
-            RCA_VERTICES_METRICS_AGGREGATOR.updateStat(
-                    ADMISSION_CONTROL_RCA_TRIGGERED, instanceDetails.getInstanceId().toString(), 1);
-
-            ResourceContext context = new ResourceContext(UNHEALTHY);
+        HeapMetric heapMetric = getHeapMetric();
+        if (!heapMetric.hasValues()) {
             return new ResourceFlowUnit<>(
-                    currentTimeMillis, context, nodeSummary, !instanceDetails.getIsMaster());
+                    currentTimeMillis,
+                    new ResourceContext(HEALTHY),
+                    nodeSummary,
+                    !instanceDetails.getIsMaster());
         }
 
-        ResourceContext context = new ResourceContext(HEALTHY);
-        return new ResourceFlowUnit<>(currentTimeMillis, context, nodeSummary);
-    }
-
-    private double getHeapBasedThreshold(double currentHeapPercent) {
-        Range range = requestSizeHeapRange.getRange(currentHeapPercent);
-        return Objects.isNull(range) ? 0 : range.getThreshold();
+        HeapRca heapRca = HeapRcaFactory.getHeapRca(heapMetric.getMaxHeap());
```

Comment:
The max heap value is going to stay the same, can we make this as a one time call?

Reply:
As PA runs outside of ES JVM, reading max heap here as it might change.